### PR TITLE
build: fix __GNU_SOURCE macro usage

### LINKED ISSUE #304: _GNU_SOURCE
The define is explicitly decelerated in multiple places. @miked-mellanox suggested to handle this define through auto tools.

### PR DESCRIPTION
**Pull request by miked-mellanox**
fixes #304 

- bump package version to 1.0
- use AC_GNU_SOURCE macro to detect and propagate __GNU_SOURCE macro

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -7,8 +7,8 @@
 AC_PREREQ([2.63])
 
 define([scm_r], esyscmd([sh -c "git rev-parse --short=7 HEAD"]))
-define([ucx_ver_major], 0)
-define([ucx_ver_minor], 1)
+define([ucx_ver_major], 1)
```
Comment (yosefe) [inline configure.ac]: should this be an another PR?

File: src/ucs/async/signal.h [hunk 2]
```diff
@@ -8,7 +8,7 @@
 #ifndef UCS_ASYNC_SIGNAL_H
 #define UCS_ASYNC_SIGNAL_H
 
-
+#include "config.h"
```
Comment (yosefe) [inline src/ucs/async/signal.h]: why is that added here?

File: src/ucs/debug/debug.h [hunk 3]
```diff
@@ -8,6 +8,9 @@
 #ifndef UCS_DEBUG_H_
 #define UCS_DEBUG_H_
 
+#ifdef HAVE_CONFIG_H
```
Comment (yosefe) [inline src/ucs/debug/debug.h]: can this be avoided?

File: configure.ac (hunk 1 above)
Comment (miked-mellanox) [inline configure.ac]: IMO - too small change for separate PR.
now version is in par w/ github milestones.

Comment (yosefe) [inline configure.ac]: so pls at least update commit message that it changes the version

Comment (miked-mellanox) [inline configure.ac]: done

**Comment by yosefe** [comment]
:+1:

**Comment by miked-mellanox** [comment]
the 1st version is 1.0, it was 0.1 (to make in sync w/ milestones versions)

**Comment by shamisp** [comment]
well, since we are only 38% in 1.0, I would keep it 0.1 and change to 1.0 once we good with all deliverables
